src/section/shared_state.rs

```diff
@@ -437,12 +417,16 @@ impl SharedState {
         let our_prefix = self.our_prefix().pushed(next_bit);
         let other_prefix = self.our_prefix().pushed(!next_bit);
 
-        let our_elders = self
-            .our_members
-            .elder_candidates_matching_prefix(&our_prefix, network_params.elder_size);
-        let other_elders = self
-            .our_members
-            .elder_candidates_matching_prefix(&other_prefix, network_params.elder_size);
+        let our_elders = self.our_members.elder_candidates_matching_prefix(
+            &our_prefix,
+            network_params.elder_size,
+            self.sections.our(),
```

Comment:
this `our` function returns with `our_elders`, right?

Reply:
It returns the current `EldersInfo` of our section.

Reply:
I'd prefer this function name to be more explicit reflecting the stuff returned, so maybe make it as `our_elders`.
Especially there are now adults/infancts within the section. (same idea to the function `our_member`, which actually also returns elders only)

However, I think this better to be done in a separate PR (if decide to do so).
So won't make this as a blocker to this PR.